hw1/02-palindrome.js

```diff
@@ -1,3 +1,36 @@
-const elem = document.querySelector('input');
+//returns true if the given number is a palindrome, false otherwise
+const isPalindrome = (num) => {
+  const original = num.toString();
 
-elem.addEventListener('input', handleInput);
+  //reverse string by splitting it into an array and reversing that, then rejoining
+  //credit: https://www.freecodecamp.org/news/how-to-reverse-a-string-in-javascript-in-3-different-ways-75e4763c68cb/
+  const reversed = original.split("").reverse().join("");
+
+  return reversed === original;
+};
+
+//Display to user whether input was a palindrome or not
+// displays invalid input if number is negative
+const handleInput = () => {
+  const input = document.getElementById("numberInput").value;
+  let outputElement = document.getElementById("response");
+  let responseText;
+
+  //Determine response to display, error on negative numbers and NaN/empty input
+  if (input === NaN || input === "" || input < 0) {
```

Comment:
The check for NaN using input === NaN is incorrect because NaN is not equal to anything, including itself. You should use isNaN(input) to check for NaN.

Inputs from an HTML form are always strings. Ensure you convert these strings to numbers before performing numerical operations like input < 0.

Reply:
Fixed, thanks.